deepdives/build-process/README.md

```diff
@@ -0,0 +1,66 @@
+# Deep Dive: The Build Process
+
+## Why?
+As you start to create more and more ambitious apps, the time you invest into a build process which automates work for you, cleans up your code, and gives you access to more features becomes more and more worth it. These tools can be intimidating to learn at first, but with a bit of pre work they can make your life a ton easier.
+
+## The Parts of the Process
+
+### Compilers
+You've already been exposed to these with SASS and LESS compilers. You can also use compilers (or sometimes call transpilers) to convert modern es6 or es7 javascript code into widely supported es5 javascript. The most recent updates to javascript has helped to smooth over some ugly parts (I'm looking at your Javascript OOP) and added a ton of new features. We'll get into how to use these new features in the exercises. Luckily, as long as we are just considering vanilla javascript, as opposed to coffescript or typescript, the de facto standard for compiling your code is:
+
+##### Babel
+![Babel](https://avatars0.githubusercontent.com/u/9637642?v=3&s=400)
+
+Babel can convert basically any javascript (including jsx!) you can imagine into es5. It does this by requireing users to specify a ```preset``` or set of presets that babel will use to do the compilation. You can also specify ```plugins``` to extend the features of babel. The way that you specify these is different based on the rest of the tools we use. You can look at the examples above to see how it is done in different setups.
+
+###### Readings
+- [Docs](https://babeljs.io/)
+- [What is Babel the Javascript Compiler](http://www.programwitherik.com/what-is-babel-the-javascript-compiler/)
+
+### Bundlers
+You may have noticed that as the clientside portion or your apps get bigger dealling with importing scripts, css, and images into your pages becomes a huge pain in the @#$. Wouldn't it be nice if we could require files like we can on our server? Well do I have good news for you, you have two good options, browserify and webpack.
+
+##### Browserify
+![Browserify](http://derickbailey.com/wp-content/uploads/2014/06/NewImage.png)
+
+Browserify aims to recreate the node module system on the client. This is great, it lets you use most node modules on the client as well as bundling your javascript into one (or a bunch of smaller if you prefer) concise file to serve. Since broswerify recreates the node require system the syntax is very familiar, but it falters when it comes to dealing with files that aren't common on server side programming such as css and images. You can shoe horn support for other files but it can get pretty messy.
```

Comment:
Browserify aims to recreate a node-style `require` system on the client. Browserify is great because it lets you bundle most serverside modules and their dependencies into files which can be used clientside; it's also great because, if you want it to, it can bundle all of your clientside JS into one concise file to serve. Since Browserify recreates the `require` system, the syntax will be familiar to you. However, Browserify falters when it comes to dealing with files which aren't common in servers -- for instance, CSS and image files. You can shoehorn things together, but it can get pretty messy.